library/psa_crypto_pake.c

```diff
@@ -244,176 +244,21 @@ psa_status_t psa_pake_setup( psa_pake_operation_t *operation,
         return( PSA_SUCCESS );
     }
     else
+#else
+    (void)operation;
+    (void)inputs;
 #endif
     status = PSA_ERROR_NOT_SUPPORTED;
 
 error:
-    psa_pake_abort( operation );
+    mbedtls_free( password );
```

Comment:
Should we set also `password_len=0` here?
Perhaps this is a minor detail because a following call to `psa_pake_set_password_key()` should set  things up properly

Reply:
`password_len` is automatic variable on stack, so its lifetime ends on return from this function.
BTW in the new driver design(operation divided into two stages: collecting inputs, computation) `mbedtls_psa_pake_setup` entry point is called not by `psa_pake_setup()`, but `psa_pake_output()` or `psa_pake_input()` (depending which one is first), so at this point we are already after `psa_pake_set_password_key()`.

Reply:
Actually I was referring to `inputs->password_len` since doing `mbedtls_free( password )` basically frees `inputs->password`. Sorry for the misunderstanding.

Reply:
`operation` object is cleared at the beginning and `operation->password_len` can be set only on success. In case of error `operation->password_len` is already `0`. 
But I think we should clear the `password` before `free`.